v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/ddl/Ddl.java

```diff
@@ -81,6 +82,25 @@ public Table apply(@Nullable String input) {
         });
   }
 
+  /**
+   * Return list of all the tables that this table refers to or depends on. This method only
+   * provides the immediate references, not the whole tree.
+   *
+   * @param tableName
+   * @return
+   */
+  public List<String> tablesReferenced(String tableName) {
+    List<String> tablesReferenced = new ArrayList<>();
+    Table table = tables.get(tableName.toLowerCase());
+    if (table.interleaveInParent() != null) {
+      tablesReferenced.add(table.interleaveInParent());
```

Comment:
Shouldnt we put interleave parent in the same set and convert the final set ot list? This could cause duplicate table names if a referenced table is also an interleave parent.